scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/constraints/activities/ActivityExpression.java

```diff
@@ -333,124 +224,44 @@ Builder getThis() {
       return this;
     }
 
-    @Override
     public @NotNull
     Builder basedOn(@NotNull ActivityExpression template) {
       type = template.type;
       startsIn = template.startRange;
       endsIn = template.endRange;
-      durationIn = template.duration;
-      startsOrEndsIn = template.startOrEndRange;
-      startsOrEndsInW = template.startOrEndRangeW;
+      durationIn = template.durationRange;
       arguments = template.arguments;
       return getThis();
     }
 
-
-    protected ActivityExpression fill(ActivityExpression template) {
-      template.type = type;
-      template.startRange = startsIn;
-      template.endRange = endsIn;
-      template.duration = durationIn;
-      template.startOrEndRange = startsOrEndsIn;
-      template.startOrEndRangeW = startsOrEndsInW;
-      template.arguments = arguments;
-      return template;
-    }
-
     /**
-     * {@inheritDoc}
+     * collect and cross-check all specified terms and construct the template
+     *
+     * creates a new template object based on a conjunction of all of the
+     * criteria specified so far in this builder
+     *
+     * multiple specifications of the same term sequentially overwrite the
+     * prior term specification
+     *
+     * the terms are checked for high level self-consistency, but it is still
+     * possible to construct predicates that will never match any activities
+     *
+     * @return a newly constructed template that matches activities meeting
+     *     the conjunction of all criteria specified to the builder
      */
-    public @NotNull
-    ActivityExpression build() {
-      final var template = new ActivityExpression();
-      fill(template);
+    public @NotNull ActivityExpression build() {
+      final var template = new ActivityExpression(
+          startsIn,
+          endsIn,
+          durationIn,
+          type,
+          nameRe,
+          arguments
+      );
       return template;
```

Comment:
Minor, can ignore: simplification
```suggestion
    public @NotNull ActivityExpression build() {
      return new ActivityExpression(
          startsIn,
          endsIn,
          durationIn,
          type,
          nameRe,
          arguments
      );
```